week-1-ts-ecosystem/w01e04-light-me-up/src/LightBulb.ts

```diff
@@ -0,0 +1,30 @@
+import { Consumable } from './PowerSource.js'
+
+export class LightBulb {
+    protected readonly powerConsumption = 20
+
+    consumeOnce (powerSource: Consumable) {
+        try {
+            powerSource.consume(this.powerConsumption)
+        } catch (err) {
+            if (err instanceof Error) {
+                console.error(err.message)
+            }
+        }
+    }
+
+    consumeTillFailure (powerSource: Consumable) {
+        let seconds = 1
+        const interval = setInterval(() => {
+            try {
+                powerSource.consume(this.powerConsumption)
+                console.log(`Glowing for ${seconds++} seconds`)
+            } catch (err) {
+                clearInterval(interval)
+                if (err instanceof Error) {
```

Comment:
Defensywnie podchodzisz z tym `instanceof` i generalnie ok.

Pamiętaj tylko, że na szczęście - błędy rzucane przez sam `JS` to zawsze instancje `Error`
Jedyny kto może namieszać to - programista

wrzucając to `throw` coś innego niż new Error() lub klasę która po nim dziedziczy.
Dlatego - dobra praktyka tego nakazuje żeby to były wyłącznie Error i dziedziczące po nim klasy.

jeśli już obsługujesz przypadek `if` to powinieneś moim zdaniem też `else` bo jeśli spodziewasz się że "walnie" tutaj coś innego niż `Error` to jeśli tak by było w istocie - to zauważ że nie będziemy o tym wiedzieli, bo nie ma żadnego `console.error`'a ani `console.log`'a dla takiego przypadku.
(nie czepiam się - bynajmniej, zwracam tylko @zawadzki-dawid Twoją uwagę na taką sytuację :)